### DISCUSSION TITLE
Identify used devices inside the REST response of api/v1/devices

### DISCUSSION THREAD

**Discussion by ConfusedMerlin** [DeviceFarmer/stf#566] [Q&A]
Greetings,

I just found out that I can identify devices, that are currently "in use" inside the json that comes back when http GET /api/v1/devices as admin user: there is a key "owner", which contains data (including the name) of the current user. Which is neat. Also, of course, there is the "using" key, which turns to true in case it is used. 

So far, so good. But then I noticed that we had one device with "automotion" in the farm too. But its owner field remains Null/None. How can identify devices  in state "automotion" AND their current user inside the json?

**Comment by denis99999** [comment]
> So far, so good. But then I noticed that we had one device with "automotion" in the farm too. But its owner field remains Null/None. How can identify devices in state "automotion" AND their current user inside the json?

What do you mean by "a device with 'automotion' in the farm"? What I do know is that a controlled device always has an owner, whether that control was taken via the UI (=> status = usage) or via the API (=> status = automation)

**Reply by ConfusedMerlin** [reply]
![2022-08-18 07 04 36 devicefarm-mobile tech rz db de 08fb9daa5898](https://user-images.githubusercontent.com/29836528/185298147-831279fc-3b58-4afc-93d8-e427bfa33dfd.png)

AutomAtion... damn it, I spend to much time creating apps for "automotion" customers I think. 
Anyway, I made a screenshot (excluding the actual owner name, because putting that here would violate that companies data privacy regulations) ... and here is, what this device's json looks like in the relevant parts. This was created using a python script that filters all keys out I don't want to show... and now, that script is not responsible for setting owner to None; the raw http GET response says the same:

RZ8T1012ZZK
model: SM-A525F
marketName: Galaxy A52
owner: None
using: False
present: True



BUUUUT... while I was writing this, more devices slipped into automation state. But this time, the Owner is actually set. Now, this dips into Bug territory, I am afraid... when only one is used in automation, it seems like the owner is not set in the http GET result, but with more being used, it is set?

![2022-08-18 07 11 31 devicefarm-mobile tech rz db de a49c6a68c3d3](https://user-images.githubusercontent.com/29836528/185299300-96067e6d-f9c1-481c-aa36-60d9e2e236a9.png)

RZ8T10129LK
model: SM-A525F
marketName: Galaxy A52
battery: {'health': 'good', 'level': 100, 'scale': 100, 'source': 'usb', 'status': 'full', 'temp': 30.5, 'voltage': 4.346}
owner: {'email': 'redacted', 'group': 'PUhs6VvgRriu6JzS+Yj/gg==', 'name': 'redacted}
using: True
present: True

R58M85Z5WVJ
model: SM-A505FN
marketName: Nuvifone
battery: {'health': 'good', 'level': 100, 'scale': 100, 'source': 'usb', 'status': 'full', 'temp': 27.8, 'voltage': 4.308}
owner: {'email': 'redacted', 'group': 'PUhs6VvgRriu6JzS+Yj/gg==', 'name': 'redacted'}
using: True
present: True

R58N113JKBL
model: SM-A505FN
marketName: Nuvifone
battery: {'health': 'good', 'level': 100, 'scale': 100, 'source': 'usb', 'status
': 'full', 'temp': 28.9, 'voltage': 4.326}
owner: {'email': 'redacted', 'group': 'PUhs6VvgRriu6JzS+Yj/gg==', 'name': 'redacted'}
using: True
present: True

R58N51KGMPV
model: SM-A505FN
marketName: Nuvifone
battery: {'health': 'good', 'level': 100, 'scale': 100, 'source': 'usb', 'status': 'full', 'temp': 28.3, 'voltage': 4.319}
owner: {'email': 'redacted', 'group': 'PUhs6VvgRriu6JzS+Yj/gg==', 'name': 'redacted'}
using: True
present: True

**Reply by denis99999** [reply]
no it's just that the serial number in question `RZ8T1012ZZK` does not appear in the `automation` state in your screen shot, moreover on my side I do not see this hypothetical bug that you describe.

**Reply by ConfusedMerlin** [reply]
> no it's just that the serial number in question `RZ8T1012ZZK` does not appear in the `automation` state in your screen shot, moreover on my side I do not see this hypothetical bug that you describe.

Oh, you are right about that. I will try to provide a better documented example in the next couple of days. Please stand by.
